circle.yml

```diff
@@ -44,11 +46,17 @@ deployment:
   docker:
     branch: master
     commands:
-      # Build small deploy container for master branch.
       - sed "s/<EMAIL>/$DOCKER_EMAIL/;s/<AUTH>/$DOCKER_AUTH/" < "resources/deploy_templates/.dockercfg.template" > ~/.dockercfg
       - |
+          VERSION=$(git describe 2>/dev/null || git rev-parse --short HEAD 2>/dev/null)
```

Comment:
`git describe` will fail only if there's no base tag (I pushed `alpha` so that it will not fail).
Its output is something like alpha-1065-g46856f4 so it is still unique for every deploy.

So yes, for now it's a lot of images but that's really not our problem. What's important is is that if you download a Docker image, you know precisely which codebase it was built from.

Once we have a set release process, there'll be less of them. And probably we won't be pushing the `dev` image any more.